server/routes.go

```diff
@@ -168,18 +153,34 @@ func GenerateHandler(c *gin.Context) {
 		return
 	}
 
-	sessionDuration := defaultSessionDuration
-	model, err := load(c, req.Model, req.Options, sessionDuration)
+	model, err := GetModel(req.Model)
```

Comment:
This is better, although I'm wondering if we can extract this into an `options` func or similar to be use both here and in `ChatHandler`?